mlir/utils/generate-test-checks.py

```diff
@@ -180,7 +180,7 @@ def process_line(line_chunks, variable_namer):
         else:
             # Otherwise, generate a new variable.
             variable = variable_namer.generate_name(ssa_name)
-            output_line += "%[[" + variable + ":.*]]"
+            output_line += "%[[" + variable + ":" + SSA_RE_STR + "]]"
```

Comment:
This would make the directive obnoxiously long. Are you hitting a lot of cases where `.*` matches too much?

Note that this script is intended to produce an initial version of the test and some manual editing is expected.

Reply:
It happens at each function definition where there are multiple memref arguments of the same type.

I am fine with keeping it as is, you are right that the tests become hard to read due to this. I will close this for now and might reopen if I think of a better option.

Reply:
In this case, I'd try looking specifically for the pattern that appears in function arguments somehow. E.g., check if two chunks starting at `%` have a common prefix that consists of the SSA name followed by something else. In that case, we could insert the more verbose regexp. This could be beneficial to more cases than just functions with memrefs.